pkg/executor/distsql.go

```diff
@@ -851,6 +860,9 @@ func (e *IndexLookUpExecutor) startIndexWorker(ctx context.Context, initBatchSiz
 			if finished {
 				break
 			}
+			if partitionTableScanExecIdx >= 0 {
+				e.dagPB.Executors[partitionTableScanExecIdx].TblScan.TableId = e.prunedPartitions[idx].GetPhysicalID()
```

Comment:
`e.dagPB.Executors` is shared across partition requests, may it introduce datarace?